src/validate.rs

```diff
@@ -145,6 +152,31 @@ async fn validate_healthchecks(
     validated
 }
 
+/// For data directory that we write to:
+/// 1. Create a tmp directory in it.
+/// 2. Change config to point to that tmp directory.
+fn create_tmp_directory(config: &mut Config, fmt: &mut Formatter) -> Option<PathBuf> {
```

Comment:
I think it'd be easier if this fs just creates a dir and returns an `Option<PathBuf>`, taking a non-`mut` `Config` ref, and then it's user responsibility to alter the `Config`.

Doing it like this is sort of like putting double responsibility on this fs, especially since the return is `Option<PathBuf>`. It would be less confusing if this fs returned `Option<()>` for instance, to force the user of this fn to read the value from the `Config`.


All this is a minor nit though, as it's private fn, and so it's contained.

Reply:
This function is designed for other `tmp` directory creations, for now `wasm` configuration will also be modified in this `fn`. Without it, the the main `validate` fn would get polluted with config manipulation which won't be pretty with `wasm`. But yes, as it is, just creating `dir` in here is better.